openssl: add patch for @1.1.1q. fixed issues #31545

William-Mou opened the pull request:
Fixes #31545 **openssl bootstrap error on Apple M1**

This issue is related to the following bug: https://github.com/openssl/openssl/issues/18720
OpenSSL has fixed it on the master branch: https://github.com/openssl/openssl/commit/f9e578e720bb35228948564192adbe3bc503d5fb

William-Mou committed:
openssl: add patch for @1.1.1q. fixed issues #31545

haampie commented on var/spack/repos/builtin/packages/openssl/package.py:
```diff
@@ -110,6 +110,10 @@ class Openssl(Package):   # Uses Fake Autotools, should subclass Package
     depends_on('nasm', when='platform=windows')
 
+    patch('https://github.com/openssl/openssl/commit/f9e578e720bb35228948564192adbe3bc503d5fb.patch?full_index=1',
+          sha256='94da5535794c8a02fa533545082e14113dad5fb6760c28b02e9f078fac14bc1c',
```
```suggestion
          sha256='3fdcf2d1e47c34f3a012f23306322c5a35cad55b180c9b6fb34537b55884645c',
```

```
curl -Lfs https://github.com/openssl/openssl/commit/f9e578e720bb35228948564192adbe3bc503d5fb.patch?full_index=1 | sha256sum
```

haampie reviewed (changes requested):
shasum doesn't match? also not sure why gitlab doesn't rebuild openssl, any clue @scottwittenburg?

haampie commented:
@spackbot run pipeline

kuberry commented:
This looks good , but I would suggest removing the `:` from the `1.1.1q:`. The fix for openssl has already been committed to master (in openssl) and should appear in the next release. https://github.com/openssl/openssl/commit/60f011f584d80447e86cae1d1bd3ae24bc13235b

kuberry commented on var/spack/repos/builtin/packages/openssl/package.py:
```diff
@@ -110,6 +110,10 @@ class Openssl(Package):   # Uses Fake Autotools, should subclass Package
 
+    patch('https://github.com/openssl/openssl/commit/f9e578e720bb35228948564192adbe3bc503d5fb.patch?full_index=1',
+          sha256='3fdcf2d1e47c34f3a012f23306322c5a35cad55b180c9b6fb34537b55884645c',
+          when='@1.1.1q:')
```
Could you change this to: `when='@1.1.1q')` ?

William-Mou commented:
@kuberry @haampie Thank you for your prompt response!
You are right, OpenSSL should be updated in the next version and solve this problem.
I have removed the ":" and corrected the checksum value.

William-Mou commented:
Hi, @haampie 
Do I have to do anything else before merging the branch?

scottwittenburg commented:
> also not sure why gitlab doesn't rebuild openssl, any clue @scottwittenburg?

I don't see that any pipeline has run on this change yet, the first one is still pending a develop pipeline.

adamjstewart reviewed (approved):
I'm able to reproduce the issue in #31545 and can confirm that this fixes the issue for me on macOS 12.4 (Apple M1 Pro) with Apple Clang 13.1.6.
